### PR TITLE
[Transforms] Fixes trailing comments emit for modules and enum declaration

### LINKED ISSUE #8045: [Transforms] Trailing comments for a namespace and enums are emitted next to var declaration
**Expected**:

```js
var m2;
(function (m2) {
    var m3;
    (function (m3) {
        /** Exported class comment*/
        var c = (function () {
            function c() {
            }
            return c;
        }());
        m3.c = c;
    })(m3 = m2.m3 || (m2.m3 = {}));
})(m2 || (m2 = {})); /* trailing dotted module comment*/
```

**Actual**:

```js
var m2; /* trailing dotted module comment*/
(function (m2) {
    var m3;
    (function (m3) {
        /** Exported class comment*/
        var c = (function () {
            function c() {
            }
            return c;
        }());
        m3.c = c;
    })(m3 = m2.m3 || (m2.m3 = {}));
})(m2 || (m2 = {}));
```

Failed tests:
* commentsModules.ts
* commentsEnums.ts

### PR DESCRIPTION
**Pull request by sheetalkamat**
This also unifies ```addVarForEnumDeclaration``` and ```addVarForModuleDeclaration```
Fixes #8045

Tests Fixed:
- tests\cases\compiler\augmentedTypesClass.ts
- tests\cases\compiler\augmentedTypesClass3.ts
- tests\cases\compiler\augmentedTypesEnum.ts
- tests\cases\compiler\augmentedTypesEnum2.ts
- tests\cases\compiler\augmentedTypesFunction.ts
- tests\cases\compiler\augmentedTypesModules.ts
- tests\cases\compiler\augmentedTypesVar.ts
- tests\cases\compiler\commentsEnums.ts
- tests\cases\compiler\commentsModules.ts

### CODE REVIEW

File: src/compiler/comments.ts [hunk 1]
```diff
@@ -117,6 +117,25 @@ namespace ts {
                     return undefined;
                 }
 
+                const node = range as Node;
+                if (node.kind === SyntaxKind.VariableStatement &&
```
Comment (mhegazy) [inline src/compiler/comments.ts]: should not this be in the callback instead?

Comment (mhegazy) [inline src/compiler/comments.ts]: //CC @rbuckton

File: src/compiler/transformers/ts.ts [hunk 2]
```diff
@@ -2261,7 +2243,7 @@ namespace ts {
 
             const statements: Statement[] = [];
             if (shouldEmitVarForEnumDeclaration(node)) {
-                addVarForEnumDeclaration(statements, node);
+                addVarForEnumOrModuleDeclaration(statements, node);
```
Comment (sandersn) [inline src/compiler/transformers/ts.ts]: I can't find `addVarForEnumOrModuleDeclaration` in `transforms` branch, and I don't see it added here. Am I missing something?

Comment (sheetalkamat) [inline src/compiler/transformers/ts.ts]: https://github.com/Microsoft/TypeScript/pull/8064/commits/27adb8c363c6a172a5a489929251f569666bd2d2#diff-c0d0e8b1528663b6cff3bb893150cec3R2382 
Renamed ```addVarForModuleDeclaration``` to ```addVarForEnumOrModuleDeclaration```

File: src/compiler/comments.ts (hunk 1 above)
Comment (sheetalkamat) [inline src/compiler/comments.ts]: As per @rbuckton suggestion made end position to be -1

**Comment by mhegazy** [comment]
:+1:

**Pull request merged**